230 customizing results output

AnirudhVIyer opened the pull request:
## Enabled urls in ResultSet output and pandas DataFrame to be clickable

## Issue ticket number and link
Closes #230 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added thorough [tests](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#testing) (when necessary).
- [x] I have added the right documentation in the [docstring](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#documenting-changes-and-new-features) and [changelog](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#changelog) (when needed)



<!-- readthedocs-preview jupysql start -->
----
:books: Documentation preview :books:: https://jupysql--274.org.readthedocs.build/en/274/

<!-- readthedocs-preview jupysql end -->

edublancas commented:
is this ready for review? looks like you ran some markdown formatted cause the CHANGELOG was edited

AnirudhVIyer commented:
Yeah, it is ready for review.

edublancas reviewed (changes requested):
please check your formatting settings, I see a lot of changes in files that should not be edited (example, the CHANGELOG and some test files)

AnirudhVIyer commented:
Will do that.

AnirudhVIyer commented:
I have added a new table with urls in the conftest file to run a few tests. Is it alright?

AnirudhVIyer commented:
Hi @edublancas can you approve the workflow checks. Thank You.

edublancas commented on src/sql/run.py:
```diff
@@ -170,7 +173,28 @@ def DataFrame(self, payload):
         "Returns a Pandas DataFrame instance built from the result set."
         import pandas as pd
 
+        "create a method to check for urls"
```
this should be a comment, not a string

```suggestion
        # create a method to check for urls"
```

please also apply it to the other comments

AnirudhVIyer replied:
Made the changes

edublancas reviewed (changes requested):
overall it looks good, just added one minor comment.

I had an issue reproducing it for pandas data frames, am I doing something wrong? 

![image](https://user-images.githubusercontent.com/989250/226507239-bc0b3bd5-64cf-418c-81bb-e46e684716d8.png)

AnirudhVIyer commented:
> overall it looks good, just added one minor comment.
> 
> I had an issue reproducing it for pandas data frames, am I doing something wrong?
> 
> ![image](https://user-images.githubusercontent.com/989250/226507239-bc0b3bd5-64cf-418c-81bb-e46e684716d8.png)

Hi Eduardo, to make the URLs clickable in a dataframe, we need to add a style function that returns a Styler object, not the original dataframe. If we return the Styler object, users won't be able to conduct further analysis on it. Therefore, I added a styling function to the returned dataframe as an attribute. Users can use this attribute to display clickable links when they want to but still retain the original dataframe for further use.

you can display the dataframe with styling like this: 

```
df.styled_func()

```

edublancas reviewed (changes requested):
ok, since getting this to work with pandas involves users knowing a bit of the implementation details, I think it's better to only keep the URL parsing the ResultSet - it's ok if the links go away when turning it into a pandas data frame

AnirudhVIyer commented:
> ok, since getting this to work with pandas involves users knowing a bit of the implementation details, I think it's better to only keep the URL parsing the ResultSet - it's ok if the links go away when turning it into a pandas data frame

Sure, I will make the required changes.

AnirudhVIyer commented:
I have removed the pandas url attribute.

edublancas commented:
great work!

AnirudhVIyer commented:
thank you!
